### PR TITLE
Add TFLiteModelImpl

### PR DESCRIPTION
**Pull request by apivovarov**
*Description of changes:*

`TFLiteModelImpl` is a wrapper on top of Tensorflow Lite which implements `IDLRModel` API

`test_tflite_model.py` contains unit tests. The unit test generates `/tmp/test_model.tflite` file and tests that DLRModel can:

1. load `/tmp/test_model.tflite` file
2. get input / output tensor names
3. run the model
4. get input values by name

### CODE REVIEW

File: python/dlr/api.py [hunk 1]
```diff
@@ -26,27 +26,41 @@ def get_output_names(self):
...
+        # Tensorflow
+        tf_model_path = _find_model_file(model_path, '.pb')
         if tf_model_path is not None:
             from .tf_model import TFModelImpl
             self._impl = TFModelImpl(tf_model_path, dev_type, dev_id)
-        else:
-            from .dlr_model import DLRModelImpl
-            self._impl = DLRModelImpl(model_path, dev_type, dev_id) 
+            return
+        # TFLite
```
Comment (yongwww) [inline python/dlr/api.py]: if we find both pb and tflite, warn user which one is used or just raise error?

File: python/dlr/tflite_model.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    Parameters
+    ----------
+    intrp : :py:class:`str`tf.lite.Interpreter`
+        TFLite Interpreter
+
+    Returns
+    -------
+    input_tensor_map : Input tensor Map which maps tensor_name to tensor_id
+    output_tensor_list  : Output tensor list. List element is pair (name, id)
```
Comment (yongwww) [inline python/dlr/tflite_model.py]:
```suggestion
    output_tensor_list : Output tensor list. List element is pair (name, id)
```

File: python/dlr/api.py (hunk 1 above)
Comment (apivovarov) [reply python/dlr/api.py]: fixed. will throw the Exception if both found

File: python/dlr/tflite_model.py (hunk 2 above)
Comment (apivovarov) [reply python/dlr/tflite_model.py]: fixed

File: Jenkinsfile [hunk 3]
```diff
@@ -135,7 +135,8 @@ def CloudInstallAndTest(cloudTarget) {
     sh """
     ls -lh *.whl
     pip3 install *.whl
-    pip3 install tensorflow
+    sudo pip3 install tensorflow
```
Comment (apivovarov) [inline Jenkinsfile]: need sudo here because it installs `toco_from_protos` to `/usr/local/bin/`

Comment (zhiics) [reply Jenkinsfile]: Trying to understand, why `toco_from_protos ` has to be installed under `/usr/local/bin`? Can we use `pip3 install --user tensorflow`?

File: python/dlr/api.py (hunk 1 above)
Comment (zhiics) [reply python/dlr/api.py]: I am okay with raising an error, but I think warning (then using TF Lite by default) might be a better option.

File: python/dlr/api.py [hunk 4]
```diff
@@ -26,27 +26,44 @@ def get_output_names(self):
...
+        if tflite_model_path is not None:
+            from .tflite_model import TFLiteModelImpl
+            self._impl = TFLiteModelImpl(tflite_model_path, dev_type, dev_id)
+            return
```
Comment (zhiics) [inline python/dlr/api.py]: Is it possible that both of them are `None`? If so, we may need to throw an error.

File: python/dlr/tflite_model.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    Parameters
+    ----------
+    intrp : :py:class:`str`tf.lite.Interpreter`
+        TFLite Interpreter
+
+    Returns
+    -------
+    input_tensor_map : Input tensor Map which maps tensor_name to tensor_id
+    output_tensor_list : Output tensor list. List element is pair (name, id)
```
Comment (zhiics) [inline python/dlr/tflite_model.py]: Why one is a map, but the other is list?

File: python/dlr/tflite_model.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+class TFLiteModelImpl(IDLRModel):
+    """
+    TFLiteModelImpl is a wrapper on top of Tensorflow Lite which implements IDLRModel API
+
+    Parameters
+    ----------
+    tflite_file : :py:class:`str`
+        Full path to TFLite file (.tflite file)
+    dev_type : :py:class:`str`
```
Comment (zhiics) [inline python/dlr/tflite_model.py]: It looks that `dev_type` and `dev_id` are TVM specific. We probably don't need to put them in the base class. This reduces confusion for users.

File: python/dlr/tflite_model.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    def get_input_names(self):
+        """
+        Get all input names
+
+        Returns
+        -------
+        out : list of :py:class:`str`
+        """
+        names = list(self._input_tensor_map.keys())
+        names.sort()
```
Comment (zhiics) [inline python/dlr/tflite_model.py]: What's the original oder? Do we need to sort them, or we can make sorting optional to the users (i.e. `get_input_name(self, sorted=False)`?

File: python/dlr/api.py (hunk 1 above)
Comment (apivovarov) [reply python/dlr/api.py]: We agreed to raise the error here

File: Jenkinsfile (hunk 3 above)
Comment (apivovarov) [reply Jenkinsfile]: Since it is in Jenkinsfile and used only by CI we agreed to keep sudo

File: python/dlr/api.py (hunk 4 above)
Comment (apivovarov) [reply python/dlr/api.py]: if both TF and TFLite files are not found then the dispatcher will use DLRModelImpl by default

File: python/dlr/tflite_model.py (hunk 5 above)
Comment (apivovarov) [reply python/dlr/tflite_model.py]: changed both input_tensor_map and output_tensor_list to OrderedDict. It allows to do lookup by tensor name and it keep tensor names in the original order returned TFLite Interpreter

File: python/dlr/tflite_model.py (hunk 6 above)
Comment (apivovarov) [reply python/dlr/tflite_model.py]: Removed `dev_type` and `dev_id` from `TFLiteImpl.init`.
If user explicitly specifies `dev_type` or/and `dev_id` for `.tflite` file then the dispatcher will log Warning and ignore `dev_type` and `dev_id` parameters

File: python/dlr/tflite_model.py (hunk 7 above)
Comment (apivovarov) [reply python/dlr/tflite_model.py]: removed `names.sort()` and use OrderedDict for `_input_tensor_map` instead

File: tests/python/unittest/test_tf_model.py [hunk 8]
```diff
@@ -59,6 +59,7 @@ def test_tf_model_on_cpu_0():
 def test_tf_model_on_gpu_0():
     test_tf_model("gpu", 0)
 
+
 if __name__ == '__main__':
     test_tf_model()
     print('All tests passed!')
```
Comment (yongwww) [inline tests/python/unittest/test_tf_model.py]: delete?

Comment (yongwww) [inline tests/python/unittest/test_tf_model.py]:
```suggestion

```

**Pull request merged**